### PR TITLE
Move shopper-side Cart E2E tests to Playwright

### LINKED ISSUE #42405: Playwright: Migrate `shopper/cart-checkout/cart.test.js`
This issue aims to migrate  `shopper/cart-checkout/cart.test.js` from Puppeteer to Playwright and is part of the [Migrate E2E tests to Playwright project](#9001).

### PR DESCRIPTION
**Pull request by tarhi-saad**
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:

Closes #42405.

### How to test the changes in this Pull Request:

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Check playwright E2E tests pass, both `block_theme` and `classic_theme`

### Changelog entry

-   [ ] Automatically create a changelog entry from the details below.

<details>

#### Significance

-   [ ] Patch
-   [ ] Minor
-   [ ] Major

#### Type

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Comment 

</details>

### CODE REVIEW

**Comment by tarhi-saad** [comment]
@tarunvijwani, there is a failing test in `templates/order-confirmation.block_theme.spec.ts`. I didn't touch this file, so maybe it failed because of another PR. Try rebasing this branch and check if the test still fails!

File: plugins/woocommerce-blocks/tests/e2e/test-1.spec.ts [hunk 1]
```diff
@@ -0,0 +1,5 @@
+import { test, expect } from '@playwright/test';
```
Comment (tarunvijwani) [inline plugins/woocommerce-blocks/tests/e2e/test-1.spec.ts]: I think these are just dummy files that got pushed mistakenly. Are we using it anywhere?

File: plugins/woocommerce-blocks/tests/e2e/tests/cart/cart-block.shopper.block_theme.side_effects.spec.ts [hunk 2]
```diff
@@ -128,4 +131,151 @@ test.describe( 'Shopper → Cart block', () => {
...
+		// Verify the "Proceed to Checkout" button is enabled after network request
+		await expect(
+			page.getByRole( 'link', { name: 'Proceed to Checkout' } )
+		).toBeEnabled();
+
+		await expect(
+			page.getByLabel(
+				`Quantity of ${ SIMPLE_VIRTUAL_PRODUCT_NAME } in your cart.`
+			)
+		).toHaveValue( '4' );
```
Comment (tarunvijwani) [inline plugins/woocommerce-blocks/tests/e2e/tests/cart/cart-block.shopper.block_theme.side_effects.spec.ts]: We increased the quantity at line 203. 
```suggestion
		).toHaveValue( '5' );
```

File: plugins/woocommerce-blocks/tests/e2e/tests/cart/cart-block.shopper.block_theme.side_effects.spec.ts [hunk 3]
```diff
@@ -128,4 +131,151 @@ test.describe( 'Shopper → Cart block', () => {
...
+		// Click on "Proceed to Checkout" button
+		await page.getByRole( 'link', { name: 'Proceed to Checkout' } ).click();
+
+		// Verify that you see the Checkout Block page
+		await expect(
+			page.getByRole( 'heading', { name: 'Checkout' } )
+		).toBeVisible();
+	} );
+
+	test( 'User can see Cross-Sells products block', async ( {
```
Comment (tarunvijwani) [inline plugins/woocommerce-blocks/tests/e2e/tests/cart/cart-block.shopper.block_theme.side_effects.spec.ts:256]: I think we need to enable cross-sells products first. This test is failing for me and I also can't see any cross-sells product in chromium: 

image

**Review by tarunvijwani — CHANGES_REQUESTED** [review]

File: plugins/woocommerce-blocks/tests/e2e/test-1.spec.ts (hunk 1 above)
Comment (nielslange) [reply plugins/woocommerce-blocks/tests/e2e/test-1.spec.ts]: I've deleted `plugins/woocommerce-blocks/tests/e2e/test-1.spec.ts` and `plugins/woocommerce-blocks/tests/e2e/test-2.spec.ts`, which indeed looked like dummy files.

File: plugins/woocommerce-blocks/tests/e2e/tests/cart/cart-block.shopper.block_theme.side_effects.spec.ts (hunk 2 above)
Comment (nielslange) [reply plugins/woocommerce-blocks/tests/e2e/tests/cart/cart-block.shopper.block_theme.side_effects.spec.ts]: Good catch, @tarunvijwani. I'll commit your suggestion.

File: plugins/woocommerce-blocks/tests/e2e/tests/cart/cart-block.shopper.block_theme.side_effects.spec.ts (hunk 3 above)
Comment (nielslange) [reply plugins/woocommerce-blocks/tests/e2e/tests/cart/cart-block.shopper.block_theme.side_effects.spec.ts:256]: Before the monorepo merge, we were using a helper plugin for setting up the Cross-Sells products. The corresponding part can be found on https://github.com/woocommerce/woocommerce-blocks/blob/4cdab7effbee82dad494a188a7026d4d8dab82a4/tests/mocks/woo-test-helper/woo-test-helper.php#L98C1-L127

I'll investigate if we can somehow reuse that existing functionality.

**Review by tarunvijwani — APPROVED** [review]
Thank you for taking over the PR, @nielslange!
It works as expected. 🎉

**Comment by nielslange** [comment]
Thanks for testing the changes again and for approving them, @tarunvijwani. 🙌

**Pull request merged**